pkg/bib/container/nxcontainer.go

```diff
@@ -0,0 +1,145 @@
+package container
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+)
+
+// Never add anything to this code that actually starts the the container
+// or execs code from it
+
+// NxContainer is a container non-executing container that runs *no* code
+// from the container and only creates/mounts it.
+type NxContainer struct {
+	id   string
+	root string
+	arch string
+}
+
+func NewNxContainer(ref string) (*NxContainer, error) {
+	output, err := exec.Command("podman", "create", ref).Output()
+	if err != nil {
+		if e, ok := err.(*exec.ExitError); ok {
+			return nil, fmt.Errorf("creating %s container failed: %w\nstderr:\n%s", ref, e, e.Stderr)
+		}
+		return nil, fmt.Errorf("creating %s container failed with generic error: %w", ref, err)
+	}
+
+	c := &NxContainer{
+		id: strings.TrimSpace(string(output)),
+	}
+	// not all containers set {{.Architecture}} so fallback
+	c.arch, err = findContainerArchInspect(c.id, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	/* #nosec G204 */
+	output, err = exec.Command("podman", "mount", c.id).Output()
+	if err != nil {
+		if err, ok := err.(*exec.ExitError); ok {
+			return nil, fmt.Errorf("mounting %s container failed: %w\nstderr:\n%s", ref, err, err.Stderr)
+		}
+		return nil, fmt.Errorf("mounting %s container failed with generic error: %w", ref, err)
+	}
+	c.root = strings.TrimSpace(string(output))
+
+	return c, err
+}
+
+func (c *NxContainer) Stop() error {
+	/* #nosec G204 */
+	if output, err := exec.Command("podman", "umount", c.id).CombinedOutput(); err != nil {
+		return fmt.Errorf("umount %s nxcontainer failed: %w\noutput:\n%s", c.id, err, output)
+	}
+
+	/* #nosec G204 */
+	if output, err := exec.Command("podman", "rm", c.id).CombinedOutput(); err != nil {
+		return fmt.Errorf("rm %s nxcontainer failed: %w\noutput:\n%s", c.id, err, output)
+	}
+	return nil
+}
```

Comment:
I do not have local branch with ctags for this one so I cannot see myself but I am hoping this is called via `defer` to ensure it never gets skipped. We do not want to run out of resources on the server.

Reply:
It is deferred already yes.